app/services/csv/lettings_log_csv_service.rb

```diff
@@ -258,6 +268,27 @@ def lettings_log_attributes
       @user.support? ? final_attributes : final_attributes - SUPPORT_ONLY_ATTRIBUTES - soft_validations_attributes
     end
 
+    def lettings_log_definitions
+      definitions = if @user.support?
+                      CsvVariableDefinition.where("log_type = 'lettings'")
+                    else
+                      CsvVariableDefinition.where("user_type = 'user' AND log_type = 'lettings'")
+                    end
```

Comment:
could this be:
`definitions = @user.csv_variable_definitions.group_by(&:variable)` and `csv_variable_definitions` is a method on the user model?

I _think_ using hash conditions in active record calls where possible is a bit more railsy as well (for example `CsvVariableDefinition.where(log_type: "lettings")`

Reply:
I've amended to return the definitions from the user model and added scopes to the CsvVariableDefinition model to grab lettings or sales records.